sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+            {
+                ExportBlock(block, method, returnType);
+            }
+            writer.WriteLine("}");
+        }
+
+        private string GetMangling(MethodDeclarationSyntax method)
+        {
+            var prettyName = EncodeName(semanticModel.GetDeclaredSymbol(method).ToDisplayString());
+            var sb = new StringBuilder(prettyName.Length);
+            foreach(char c in prettyName)
+            {
+                if (char.IsLetterOrDigit(c) || c == '.' || c == '_')
+                {
+                    sb.Append(c);
+                }
+                else if (char.IsSeparator(c))
+                {
+                    // Ignore it
+                }
+                else if(c == ',')
+                {
+                    sb.Append('.');
+                }
+                else
+                {
+                    sb.Append('$');
+                }
+            }
+            return sb.ToString();
+        }
+        private bool IsTooClomplexForMLIROrTheCFG(MethodDeclarationSyntax method)
+        {
+            var symbol = semanticModel.GetDeclaredSymbol(method);
+            if (symbol.IsAsync)
+            {
+                exporterMetrics.AddUnsupportedFunction(SyntaxKind.AsyncKeyword);
+                return true;
+            }
+            foreach (var node in method.DescendantNodes())
+            {
+                if (unsupportedSyntaxes.Contains(node.Kind()))
+                {
+                    exporterMetrics.AddUnsupportedFunction(node.Kind());
+                    return true;
+                }
+            }
+            exporterMetrics.AddSupportedFunction();
+            return false;
+        }
+
+        private void CreateEntryBlock(MethodDeclarationSyntax method)
+        {
+            writer.WriteLine($"^entry {GetArgumentsString(method)}:");
+            foreach (var param in method.ParameterList.Parameters)
+            {
+                if(string.IsNullOrEmpty(param.Identifier.ValueText))
+                {
+                    // An unnamed parameter cannot be used inside the function
+                    continue;
+                }
+                var id = OpId(param);
+                writer.WriteLine($"%{id} = cbde.alloca {MLIRType(param)} {GetLocation(param)}");
+                writer.WriteLine($"cbde.store %{EncodeName(param.Identifier.ValueText)}, %{id} : memref<{MLIRType(param)}> {GetLocation(param)}");
+            }
+            writer.WriteLine("br ^0");
+            writer.WriteLine();
+        }
+
+        private bool HasNoReturn(MethodDeclarationSyntax method)
+        {
+            return semanticModel.GetTypeInfo(method.ReturnType).Type.SpecialType == SpecialType.System_Void;
+        }
+
+        private void ExportBlock(Block block, MethodDeclarationSyntax parentMethod, string functionReturnType)
+        {
+            writer.WriteLine($"^{BlockId(block)}: // {block.GetType().Name}"); // TODO: Block arguments...
```

Comment:
are you going to address this TODO?

Reply:
It will probably naturally come when handling `A ? B : C`, so no need to keep this TODO for now.